### ISSUE TITLE
getApplicationDocumentsDirectory() Still With A Different Context Directory from the .send Method

### ISSUE THREAD

**Issue by psygo** [sidlatau/flutter_email_sender#35]
This is basically me trying to reopen Issue #10.

I'm using version `2.2.2` with Android 9 (API 28) on a Nexus 6 and the `FlutterEmailSender` is still getting the directory context from a different place, not `/data/user/0/...`.

@skaletaPL mentions a workaround by using `getTemporaryDirectory()` instead of `getApplicationDocumentsDirectory()`, but I don't think this should be mandatory. Or, if it should, it could be stated in the package description somewhere to warn users.

**Comment by chrislaurie-bs** [comment]
I'm having a similar issue. My local storage class returns a file path of:

/data/user/0/...

Which I pass into attachmentPaths as is.

The sending of the email fails with this error:

PlatformException(error, Failed to find configured root that contains /data/data/...

On a Samsung Galaxy 7

**Comment by kiostudio** [comment]
Hey guys, facing the same issue here. Here's my workaround by copying the file to the temp directory :

```
Directory directory = await getTemporaryDirectory();
String path = directory.path;
File newImage = await image.copy('${path}/filename.jpg');
final_path = newImage.path;
```

**Comment by sidlatau** [comment]
Fixed in 5.1.0
